cmd/endpoint.go

```diff
@@ -133,6 +136,14 @@ func NewEndpoint(arg string) (ep Endpoint, e error) {
 		}
 	} else {
 		u = &url.URL{Path: path.Clean(arg)}
+
+		// Check if the arg is a localhost or an IP without a scheme.
+		if strings.Contains(arg, "localhost") || (checkIP4(arg) == true) {
+			if !((u.Scheme == "http" || u.Scheme == "https") &&
+				u.User == nil && u.Opaque == "" && u.ForceQuery == false && u.RawQuery == "" && u.Fragment == "") {
+				return ep, fmt.Errorf("invalid URL endpoint format: missing scheme http or https")
```

Comment:
ok really great catch. Actually once the control flow gets here, it means that the string was not splittable as a regular URL with scheme and host etc but localhost or the ip is erroneously taken as a regular filesystem path. Infact it creates a local directory with the ip address or localhost.  
So removed all this and send this PR again.